linera-chain/src/manager.rs

```diff
@@ -549,6 +550,21 @@ impl ChainManager {
     fn is_super(&self, owner: &Owner) -> bool {
         self.ownership.super_owners.contains_key(owner)
     }
+
+    fn update_locked(&mut self, new_locked: Certificate) {
+        self.locked = Some(new_locked);
+        self.locked_blobs = self.proposed_blobs.clone();
+    }
+
+    fn update_proposed(&mut self, new_proposed: BlockProposal) {
+        self.proposed_blobs = new_proposed
+            .blobs
+            .clone()
+            .into_iter()
+            .map(|blob| (blob.id(), blob))
+            .collect();
```

Comment:
Ah, right, they are already in the proposal!
Why have a separate field at all then? _Especially_ in the `ChainInfo`!

Reply:
Alternatively, you could `drain(..)` the `blobs` from `new_proposal`, if you want to keep them separately?

Reply:
Can't we have a method, instead of a field, `fn proposed_blobs(&self)` which would return a reference to the `self.proposed.unwrap_or_default()....` ?

I don't like draining the `new_proposal` (`self.proposed`) as it will mean the consumer of `self.proposed` would need to "reconstruct" the proposal (put the blobs back to it).